tests/v2/test_waku_store.nim

```diff
@@ -0,0 +1,83 @@
+import unittest, options, tables, sets, sequtils
+import chronos, chronicles
+import utils,
+       libp2p/errors,
+       libp2p/switch,
+       libp2p/protobuf/minprotobuf,
+       libp2p/stream/[bufferstream, connection],
+       libp2p/crypto/crypto,
+       libp2p/protocols/pubsub/floodsub,
+       libp2p/protocols/pubsub/rpc/message,
+       libp2p/multistream,
+       libp2p/transports/transport,
+       libp2p/transports/tcptransport
+import ../../waku/protocol/v2/[waku_protocol2, waku_store, filter]
+
+import ../test_helpers
+
+procSuite "Waku Store":
+
+  test "encoding and decoding StoreRPC":
+    let
+      peer = PeerInfo.init(PrivateKey.random(ECDSA, rng[]).get())
+      msg = Message.init(peer, @[byte 1, 2, 3], "topic", 3, false)
+
+      rpc = StoreRPC(query: @[HistoryQuery(topics: @["foo"])], response: @[HistoryResponse(messages: @[msg])])
+
+    let buf = rpc.encode()
+
+    let decode = StoreRPC.init(buf.buffer)
+
+    check:
+      decode.isErr == false
+      decode.value == rpc
+
+  asyncTest "handle query":
+    let 
+      proto = WakuStore.init()
+      filter = proto.filter()
+
+    var filters = initTable[string, Filter]()
+    filters["test"] = filter
+
+    let
+      peer = PeerInfo.init(PrivateKey.random(ECDSA, rng[]).get())
+      msg = Message.init(peer, @[byte 1, 2, 3], "topic", 3, false)
+      msg2 = Message.init(peer, @[byte 1, 2, 3], "topic2", 4, false)
+
+    filters.notify(msg)
```

Comment:
I don't understand what this is or how it is supposed to work. The only mention of filter in spec is https://specs.vac.dev/specs/waku/waku-v2.html#protobuf-1 under "content filter", but that seems like it is part of the other filter PR.

It should get messages from relay handlers, should it not?

Reply:
This is just an internal pubsub filter. Where node protocols can listen to messages, aka the relay handler would post into the filter.